pkg/gateway/model_build_targets_test.go

```diff
@@ -303,6 +313,134 @@ func Test_Targets(t *testing.T) {
 				},
 			},
 		},
+		{
```

Comment:
It would be great if we could have endpoints with different ports in the test case, since that is the primary use case we want to support by this PR